Add difficulty meta to notebooks

anarafaelaferro opened the pull request:
We want to be able to filter the notebooks based on specific attributes (e.g. the minimum tier, already added; and the difficulty).

https://www.figma.com/design/GGphRwQc9yJm67wWLNDu5a/Redesign-Gallery-for-Notebooks?node-id=876-22033&t=d0ASWwUikeQGD9sG-0

Where tags included "starter/beginner/advanced", added `difficulty="beginner"/"advanced"`. Should I assume it's ok to add `difficulty="intermediate"` to the remaining?

kesmit13 commented:
Can you run the pre-commit checks and push again? It looks like one of the files needs a formatting change.

anarafaelaferro commented:
> Can you run the pre-commit checks and push again? It looks like one of the files needs a formatting change.

Yes! I'm trying to, just trying to fix some issues (I think something with my setup).

anarafaelaferro commented:
@kesmit13 checks are passing now. Could you give me some input on this?

> Should I add difficulty="intermediate" to the [notebooks that didn't have a difficulty tag]? Or do we need to analyze one by one?

kesmit13 commented:
> @kesmit13 checks are passing now. Could you give me some input on this?
> 
> > Should I add difficulty="intermediate" to the [notebooks that didn't have a difficulty tag]? Or do we need to analyze one by one?

I'm not sure who requested this flag, so I don't know if it should be required or not. If it is required, setting the remaining ones to intermediate would likely be necessary.

anarafaelaferro commented:
> > @kesmit13 checks are passing now. Could you give me some input on this?
> > > Should I add difficulty="intermediate" to the [notebooks that didn't have a difficulty tag]? Or do we need to analyze one by one?
> 
> I'm not sure who requested this flag, so I don't know if it should be required or not. If it is required, setting the remaining ones to intermediate would likely be necessary.

I'm not sure who originally made this request, but the goal is to add filters to the website on singlestore.com/spaces to make it easier to find relevant notebooks. If the difficulty is missing, the notebook won't be listed when a difficulty filter is selected. So, I'm trying to understand if difficulty=intermediate is accurate on all those remaining notebooks, or if we'll need to evaluate each case one by one. It won't "break" anything, just want to make sure they're all tagged correctly.

kesmit13 commented:
> > > @kesmit13 checks are passing now. Could you give me some input on this?
> > > > Should I add difficulty="intermediate" to the [notebooks that didn't have a difficulty tag]? Or do we need to analyze one by one?
> > 
> > 
> > I'm not sure who requested this flag, so I don't know if it should be required or not. If it is required, setting the remaining ones to intermediate would likely be necessary.
> 
> I'm not sure who originally made this request, but the goal is to add filters to the website on singlestore.com/spaces to make it easier to find relevant notebooks. If the difficulty is missing, the notebook won't be listed when a difficulty filter is selected. So, I'm trying to understand if difficulty=intermediate is accurate on all those remaining notebooks, or if we'll need to evaluate each case one by one. It won't "break" anything, just want to make sure I don't tag them incorrectly.

I guess if it's not beginner or advanced, it must be intermediate, right? :) There aren't any other valid possibilities. It might be best to just make it required and add it to the remaining notebooks.

anarafaelaferro commented:
- Added `difficulty=intermediate` to untagged notebooks
- Updated test so difficulty is mandatory

![Screenshot from 2024-05-28 19-02-48](https://github.com/singlestore-labs/spaces-notebooks/assets/17345789/9a5dcff5-6217-476a-a76a-372a8b41182e)
![Screenshot from 2024-05-28 19-02-43](https://github.com/singlestore-labs/spaces-notebooks/assets/17345789/7bb370fa-530d-4715-8afe-3fec0d642218)
